mlx/ops.cpp

```diff
@@ -445,127 +445,81 @@ array expand_dims(
   return reshape(a, out_shape, s);
 }
 
-array slice(
-    const array& a,
+// Slice helper
+namespace {
+
+inline auto normalize_slice(
+    std::vector<int> shape,
     std::vector<int> start,
     std::vector<int> stop,
-    std::vector<int> strides,
-    StreamOrDevice s /* = {} */) {
-  if (start.size() != a.ndim() || stop.size() != a.ndim() ||
-      strides.size() != a.ndim()) {
-    std::ostringstream msg;
-    msg << "[slice] Invalid number of indices or strides for "
-        << "array with dimension " << a.ndim() << ".";
-    throw std::invalid_argument(msg.str());
-  }
+    std::vector<int> strides) {
```

Comment:
Why not make these references and update them in place. I think they are already copied into the `slice` and `slice_update`.

Reply:
Good point